transport/transporters/rabbitmq/factory.go

```diff
@@ -0,0 +1,200 @@
+/*
+  Copyright 2019 Nextdoor.com, Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package rabbitmq
+
+import (
+	"net/url"
+	"os"
+
+	"github.com/NeowayLabs/wabbit"
+	"github.com/NeowayLabs/wabbit/amqp"
+	"github.com/Nextdoor/pg-bifrost.git/shutdown"
+	"github.com/Nextdoor/pg-bifrost.git/stats"
+	"github.com/Nextdoor/pg-bifrost.git/transport"
+	"github.com/Nextdoor/pg-bifrost.git/transport/batch"
+	"github.com/Nextdoor/pg-bifrost.git/transport/transporters/rabbitmq/transporter"
+	"github.com/cevaris/ordered_map"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	logger = logrus.New()
+	log    = logger.WithField("package", "rabbitmq")
+)
+
+type config struct {
+	host      string
+	username  string
+	password  string
+	vhost     string
+	exchange  string
+	batchSize int
+}
+
+func init() {
+	logger.SetOutput(os.Stdout)
+	logger.SetLevel(logrus.InfoLevel)
+}
+
+// New creates a rabbitmq transport which has rabbitmq batching and rabbitmq transporting, connected by channels
+func New(shutdownHandler shutdown.ShutdownHandler,
+	txnsWritten chan<- *ordered_map.OrderedMap, // Map of <transaction:progress.Written>
+	statsChan chan stats.Stat,
+	workers int,
+	inputChans []<-chan transport.Batch,
+	transportConfig map[string]interface{}) []*transport.Transporter {
+
+	conf := getConfig(transportConfig)
+
+	conn := createConnection(conf.host, conf.username, conf.password, conf.vhost, log)
+	channel, err := conn.Channel()
+	if err != nil {
+		log.WithError(err).Fatal("Could not open channel")
+	}
+	defer channel.Close()
+	err = channel.ExchangeDeclare(
+		conf.exchange, // name of the exchange
+		"topic",       // type
+		wabbit.Option{
+			"durable":  true,
+			"delete":   false,
+			"internal": false,
+			"noWait":   false,
+		},
+	)
+	log.WithField("exchangeName", conf.exchange).WithError(err).Fatal("Could not declare exchange")
```

Comment:
Should this be wrapped in an `err != nil` conditional?

Reply:
Yes, I have that fixed locally and will be adding test coverage to catch similar errors in the future.